File: solutions/_1456_maximum_number_of_vowels_in_a_substring_of_given_length/solution.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = set(["a", "e", "i", "o", "u"])
        max_vowels = 0
        current_vowels = 0
        for i in range(len(s)):
            if s[i] in vowels:
                current_vowels += 1
            if i >= k and s[i - k] in vowels:
                current_vowels -= 1
            max_vowels = max(max_vowels, current_vowels)
        return max_vowels


# https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-given-length/

File: solutions/_1456_maximum_number_of_vowels_in_a_substring_of_given_length/solution.java

package _1456_maximum_number_of_vowels_in_a_substring_of_given_length;

import java.util.HashSet;

class Solution {
    public int maxVowels(String s, int k) {
        var vowels = new HashSet<Character>();
        for (char c : "aeiou".toCharArray())
            vowels.add(c);

        int max = 0;
        int count = 0;

        for (int i = 0; i < s.length(); i++) {
            if (vowels.contains(s.charAt(i)))
                count++;
            if (i >= k && vowels.contains(s.charAt(i - k)))
                count--;
            if (count > max)
                max = count;
        }

        return max;
    }
}

/*
 * https://leetcode.com/problems/maximum-number-of-vowels-in-a-substring-of-
 * given-length/
 */